cmd/zinject/zinject.c

```diff
@@ -1136,6 +1216,18 @@ main(int argc, char **argv)
 			return (1);
 		}
 
+		if (dva_mask != 0) {
+			if (error == EACCES || error == EINVAL) {
+				(void) fprintf(stderr, "the '-c' option may "
+				    "not be used with logical data errors "
+				    "'decrypt' and 'decompress'\n");
+				libzfs_fini(g_zfs);
+				return (1);
+			}
+
+			record.zi_dvas = dva_mask;
```

Comment:
My goal by doing it this way was to have a clear differentiation between logical errors and DVA child errors. I think this is important because some errors like `decompress` and `decrypt` failures don't really apply to individual DVAs.

Reply:
What if I used `#define ZI_DVAS_NONE   (-1UL)`? This could help clear up the meaning of the `-1` in the code while preserving the separation I mentioned.